ruby/bindings/InitRubyBindings.cpp

```diff
@@ -467,15 +471,52 @@ def safe_puts(message=nil, opts=nil)
   end
 end
 
-require 'logger'
 require 'rbconfig'
 
-$logger = Logger.new(STDOUT)
-#$logger.level = Logger::ERROR
-$logger.level = Logger::WARN
+# require 'logger'
+# $logger = Logger.new(STDOUT)
+# $logger.level = Logger::DEBUG #WARN
+# Mimic ruby Logger, but forward to OpenStudio, so we don't have to worry about setting the LogLevel
+class ConvenienceLogger
+
+  attr_accessor :level
+
+  def initialize(log_level)
+    @level = log_level
+  end
+
+  def trace?
+    @level <= OpenStudio::Trace
+  end
+
+  def debug(msg)
+    OpenStudio::logFree(OpenStudio::Debug, "ruby", msg)
+  end
+
+  def info(msg)
+    OpenStudio::logFree(OpenStudio::Info, "ruby", msg)
+  end
+
+  def warn(msg)
+    OpenStudio::logFree(OpenStudio::Warn, "ruby", msg)
+  end
+
+  def error(msg)
+    OpenStudio::logFree(OpenStudio::Error, "ruby", msg)
+  end
+
+  def fatal(msg)
+    OpenStudio::logFree(OpenStudio::Fatal, "ruby", msg)
+  end
+
+end
+$logger = ConvenienceLogger.new(OpenStudio::Logger.instance.standardOutLogger.logLevel.get)
```

Comment:
A clever way to keep the "$logger" code below working, while respecting the user's choice in terms of log level. PReviosuly the C++ CLI was lacking the ability to debug this stuff because the loglevel was set in stone to WARN.